patroni/config.py

```diff
@@ -45,7 +46,11 @@ def __init__(self, config_file=None, config_env=None):
         self._config_file = None if config_env else config_file
         self._modify_index = -1
         self._dynamic_configuration = {}
-        self._local_configuration = yaml.safe_load(config_env) if config_env else self._load_config_file()
+        if config_env:
+            self._local_configuration = yaml.safe_load(config_env)
+        else:
+            self.__environment_configuration = self._build_environment_configuration()
```

Comment:
You can call _build_environment_configuration directly inside the _load_config_file. Make it the staticmethod by passing the self.__config_file as a parameter.